partcad/tests/unit/test_project.py

```
@@ -57,6 +72,48 @@ def test_project_import_git2():
+        patch.object(ctx, "_load_connectivity_cache", return_value=MagicMock()) as load_connectivity_mock, \
+        patch('time.time', return_value=now), \
+        patch.object(pc.user_config, "offline", offline), \
+        patch.object(pc.user_config, "force_update", force_update), \
+        patch.object(ctx, 'connect_state', {"last_checked": last_checked, "is_connected": is_connected}):
+
+        factory = pc.ProjectFactoryGit(ctx, None, test_config_import_git)
+        assert factory is not None
+
+        # Scenario 1: Offline mode should not perform connectivity check
+        if offline:
+            assert ctx.is_connected() == False
```

Ruff (0.8.2)

Avoid equality comparisons to `False`; use `if not ctx.is_connected():` for false checks

Replace with `not ctx.is_connected()`

(E712)